src/main/java/com/alex/eshop/repository/CategoryRepository.java

```diff
@@ -9,5 +9,5 @@
 @Repository
 public interface CategoryRepository extends JpaRepository<Category, Long> {
     @Query("select count(i.category.id) from Item i where i.category.id =:id")
-    Long countCategoryByItems(Long id);
+    Long countItemsByCategory(Long id);
```

Comment:
More logical jpql will be `select count(i) from Item i where i.category.id =:id`

Or simpler, will be adding `long countByCategoryId(Long categoryId);` to ItemRepository.